src/autopas/tuning/tuningStrategy/decisionTreeTuning/DecisionTreeTuning.cpp

```diff
@@ -0,0 +1,168 @@
+/**
+ * @file DecisionTreeTuning.cpp
+ * @author Abdulkadir Pazar
+ * @date 20.06.24
+ */
+
+#include "DecisionTreeTuning.h"
+
+#include <Python.h>
+
+#include <iostream>
+#include <sstream>
+
+namespace autopas {
+
+// Constructor
+DecisionTreeTuning::DecisionTreeTuning(const std::set<Configuration> &searchSpace, const std::string &modelFileName)
+    : _configurations(searchSpace), _modelFileName(modelFileName), _pFunc(nullptr) {
+  // Initialize Python Interpreter
+  Py_Initialize();
+  loadScript();
+}
+
+// Destructor
+DecisionTreeTuning::~DecisionTreeTuning() {
+  // Finalize Python Interpreter
+  if (_pFunc) {
+    Py_DECREF(_pFunc);  // Clean up the reference to _pFunc
+  }
+  Py_Finalize();
+}
+
+void DecisionTreeTuning::loadScript() {
+  // Add the current directory to the Python path
+  PyObject *sysPath = PySys_GetObject((char *)"path");
```

Comment:
We shouldn't have to require the python scripts to be in the same folder as the one the execution is run from.

Reply:
What would be your suggestion? Should they remain in src/autopas/tuning/tuningStrategy/decisionTreeTuning/ and not be moved to the execution directory when building?

Reply:
Implemented.